Arelle/Arelle
# Modernize ESEF conformance suite scripts and add integration tests

#### Reason for change
In an attempt to modernize Arelle and provide easier to use testing capabilities, I have modernized the ESEF conformance suite scripts so that anyone with a python environment can easily run, verify and test ESEF code. 

Additionally I have setup the foundation of integration testing that can be used to run the ESEF conformance suite for a potential CI feature in the future.

#### Steps to Test
- Run `python -m scripts.conformance_suites.run_esef_conformance_suite` from the main Arelle directory to run the standard script and verify successful completion. It should produce a log file and an xlsx of the results.
- Run `pytest --disable-warnings tests` from the main Arelle directory to run the integration tests and verify successful completion. These tests will not pass at the moment. More work will be done to make sure that Arelle conforms 100% to the 2021 conformance suite.

**review**:
@Arelle/arelle
